module/zcommon/zfs_fletcher_aarch64_neon_impl.S

```diff
@@ -0,0 +1,113 @@
+/*
+ * Optimize the neon implementation which is generated by GCC12
+ * or Clang by the commit: https://github.com/openzfs/zfs/pull/14234
+
+ * Due to the compiler limitation, the GCC version before 12 will
+ * produce not optimized code for Aarch64, so use the asm function
+ * to accelerate the performance here.
+ * Copyright (C) 2022 Kevin Zhao.
+ *
+ * Authors:
+ *	Kevin Zhao <kevin.zhao@linaro.org>
```

Comment:
We probably should mark this as a machine generated file (possibly with some cleanup) and put the original C file in `contrib/`. Then just document that this file was generated by Clang from that file.

Something like this could go into contrib/:

https://gcc.godbolt.org/z/1Tr6Wev4M

My guess would be something like `contrib/zcommon/zfs_fletcher_aarch64_neon_impl.c` would work. There is no need to add this to the build system since we want the build system to use the assembly here.